src/lib.rs

```diff
@@ -113,35 +166,56 @@ pub struct Peripheral {
 }
 
 impl Peripheral {
-    fn parse(tree: &Element) -> Peripheral {
-        assert_eq!(tree.name, "peripheral");
-
-        Peripheral {
-            name: try!(tree.get_child_text("name")),
-            group_name: tree.get_child_text("groupName"),
-            description: tree.get_child_text("description"),
-            base_address: try!(parse::u32(try!(tree.get_child("baseAddress")))),
-            interrupt: tree.children
-                .iter()
-                .filter(|t| t.name == "interrupt")
-                .map(Interrupt::parse)
-                .collect::<Vec<_>>(),
-            registers: tree.get_child("registers")
-                .map(
-                    |rs| {
-                        rs.children
-                            .iter()
-                            .filter_map(Register::parse)
-                            .collect()
+    fn parse(tree: &Element) -> Result<Peripheral> {
+        if tree.name != "peripheral" {
+            bail!("peripheral not peripheral")
```

Comment:
I was wondering about this one, it checks if an element of `<peripherals>` is a `<peripheral>`, maybe it should be reworded to be `"Element is <{tree.name}>, expected <peripheral>"` and chained together with `"While parsing peripheral #x of <peripherals>"`